docs/spanner.md

```diff
@@ -0,0 +1,190 @@
+# Cloud Spanner + OnlineBoutique
+
+This guide contains instructions for overriding the default in-cluster `redis` database for `cartservice` with Cloud Spanner.
+
+Run these steps after the **Create a GKE cluster** step and before the **Deploy the sample app to the cluster** in the README Quickstart (GKE).
+
+## Steps to use Spanner for your cluster
+
+All commands below should be run from the root directory of this project.
+Run the commands in the same session to keep environmental variables consistent.
+
+### 0. Environmental Variables
+
+Choose your desired project, region, and zone.
+
+```sh
+PROJECT_ID="<your-project-id>"
+REGION="<your-GCP-region>"      # e.g. us-central1
+ZONE="<your-GCP-zone>"          # e.g. us-central1-b
+```
+
+### 1. Create a GKE cluster with Workload Identity enabled
+
+These commands are similar as what is listed in the README Quickstart (GKE),
+but include a flag to [enable workload identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable).
+This will allow the GKE service account to query Spanner.
+
+GKE autopilot mode (see [Autopilot
+overview](https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-overview)
+to learn more):
+
+```sh
+gcloud container clusters create-auto onlineboutique \
+    --project=${PROJECT_ID} --region=${REGION}
+```
+
+GKE Standard mode:
+
+```sh
+gcloud container clusters create onlineboutique \
+    --project=${PROJECT_ID} --zone=${ZONE} \
+    --workload-pool=${PROJECT_ID}.svc.id.goog \
+    --machine-type=n2-standard-2 --num-nodes=4
+```
+
+### 2. Enable Cloud Spanner on your project
+
+```sh
+gcloud services enable spanner.googleapis.com --project=${PROJECT_ID}
+```
+
+### 3. Create a Cloud Spanner instance, database, and table
+
+Note: See the documentation to list [available Spanner configuration names](https://cloud.google.com/spanner/docs/getting-started/set-up#run_the_gcloud_tool), or run `gcloud spanner instance-configs list --project=$PROJECT_ID`
+
+```sh
+SPANNER_REGION_CONFIG="<your-spanner-region-config-name>" # e.g. "regional-us-east5"
```

Comment:
I've updated the example to `"regional-us-east5"` because that's [where the free instance is available](https://cloud.google.com/spanner/docs/free-trial-instance#:~:text=one%20of%20the%20following%20regional%20instance).